src/game/models/asteroid.rs

```diff
@@ -144,6 +170,81 @@ impl Asteroid {
             on_screen: false,
         }
     }
+
+    pub fn can_split(&self) -> bool {
+        self.shape.len() > 10
+    }
+
+    pub fn split<P: Positioned>(&mut self, other: &P) -> Vec<Asteroid> {
+        self.normalize_rotation();
+        //let base_speed = self.vel();
+        let index_nearest = self.index_nearest_point(other);
+        let num_pieces = 3;
```

Comment:
Adding a little bit of randomness to how it breaks up (`num_pieces`) could really make it look cool too. Maybe like a value between 2-5?